doc/ApiDocGeneration/Generate-Api-Docs.ps1

```diff
@@ -0,0 +1,305 @@
+
+    Write-Verbose "Include XML Files"
+    & "$($BinDirectory)/PopImport/popimport.exe" -f "$($BinDirectory)/$($SafeName)/dll-docs"
+
+    Write-Verbose "Produce ECMAXML"
+    & "$($BinDirectory)/mdoc/mdoc.exe" update -fx "$($BinDirectory)/$($SafeName)/dll-docs" -o "$($BinDirectory)/$($SafeName)/dll-xml-output" --debug -lang docid -lang vb.net -lang fsharp --delete
+
+    Write-Verbose "Generate YAML"
+    & "$($BinDirectory)/ECMA2Yml/ECMA2Yaml.exe" -s "$($BinDirectory)/$($SafeName)/dll-xml-output" -o "$($BinDirectory)/$($SafeName)/dll-yaml-output"
+
+    Write-Verbose "Provision DocFX Directory"
+    & "$($BinDirectory)/docfx/docfx.exe" init -q -o "$($BinDirectory)/$($SafeName)/docfx-output/docfx_project"
+
+    Write-Verbose "Copy over Package ReadMe"
+    $PkgReadMePath = "$($RepoRoot)/sdk/$($PackageLocation)/README.md"
+    if ([System.IO.File]::Exists($PkgReadMePath))
+    {
+        Copy-Item $PkgReadMePath -Destination "$($BinDirectory)/$($SafeName)/docfx-output/docfx_project/api/index.md" -Force
+        Copy-Item $PkgReadMePath -Destination "$($BinDirectory)/$($SafeName)/docfx-output/docfx_project/index.md" -Force
+    }
+    else
+    {
+        New-Item "$($BinDirectory)/$($SafeName)/docfx-output/docfx_project/api/index.md" -Force
+        Add-Content -Path "$($BinDirectory)/$($SafeName)/docfx-output/docfx_project/api/index.md" -Value "This Package Contains no Readme."
+        Copy-Item "$($BinDirectory)/$($SafeName)/docfx-output/docfx_project/api/index.md" -Destination "$($BinDirectory)/$($SafeName)/docfx-output/docfx_project/index.md"-Force
+        Write-Verbose "Package ReadMe was not found"
+    }
+
+    Write-Verbose "Copy over generated yml and other assets"
+    Copy-Item "$($BinDirectory)/$($SafeName)/dll-yaml-output/*"-Destination "$($BinDirectory)/$($SafeName)/docfx-output/docfx_project/api" -Recurse
+    Copy-Item "$($RepoRoot)/doc/ApiDocGeneration/assets/docfx.json" -Destination "$($BinDirectory)/$($SafeName)/docfx-output/docfx_project" -Recurse -Force
+    New-Item -Path "$($BinDirectory)/$($SafeName)/docfx-output/docfx_project" -Name templates -ItemType directory
+    Copy-Item "$($RepoRoot)/doc/ApiDocGeneration/templates/**" -Destination "$($BinDirectory)/$($SafeName)/docfx-output/docfx_project/templates" -Recurse -Force
+
+    Write-Verbose "Create Toc for Site Navigation"
+    New-Item "$($BinDirectory)/$($SafeName)/docfx-output/docfx_project/toc.yml" -Force
+    Add-Content -Path "$($BinDirectory)/$($SafeName)/docfx-output/docfx_project/toc.yml" -Value "- name: $ArtifactName`r`n  href: api/`r`n  homepage: api/index.md"
+
+    Write-Verbose "Build Doc Content"
+    & "$($BinDirectory)/docfx/docfx.exe" build "$($BinDirectory)/$($SafeName)/docfx-output/docfx_project/docfx.json"
+
+    Write-Verbose "Copy over site Logo"
+    Copy-Item "$($RepoRoot)/doc/ApiDocGeneration/assets/logo.svg" -Destination "$($BinDirectory)/$($SafeName)/docfx-output/docfx_project/_site" -Recurse -Force
+}
+
+Function Build-DocIndex
+{
```

Comment:
Does Build-Docs and Build-DocIndex share any code? If not I would put them into separate scripts. It will help keep concerns separated.

Reply:
Looks like they share Get-DocFx perhaps that could be independent step and the path could be passed in.

Reply:
Well we can do that but that will defeat the idea of being able to reproduce the docs locally as the steps are now broken up and one needs to download the tools first separately before ruining the script. Here you can just call the function with right parameters to generate docs as well as doc index.

Reply:
Separated into different script.